packages/toolkit/src/query/core/buildThunks.ts

```diff
@@ -435,37 +448,56 @@ In the case of an unhandled error, no tags will be "provided" or "invalidated".`
       endpointName: EndpointName,
       arg: any,
       options: PrefetchOptions
-    ): ThunkAction<void, any, any, AnyAction> =>
-    (dispatch: ThunkDispatch<any, any, any>, getState: () => any) => {
+    ): ThunkAction<PrefetchActionCreatorResult, any, any, AnyAction> =>
+    (dispatch: ThunkDispatch<any, any, AnyAction>, getState: () => any) => {
       const force = hasTheForce(options) && options.force
       const maxAge = hasMaxAge(options) && options.ifOlderThan
 
-      const queryAction = (force: boolean = true) =>
-        (api.endpoints[endpointName] as ApiEndpointQuery<any, any>).initiate(
-          arg,
-          { forceRefetch: force }
+      const dispatchPrefetchRequest = (
+        forceRefetch: boolean = true
+      ): PrefetchActionCreatorResult => {
+        const initiateOutput = dispatch(
+          (api.endpoints[endpointName] as ApiEndpointQuery<any, any>).initiate(
+            arg,
+            { forceRefetch, prefetch: options || true }
+          )
         )
-      const latestStateValue = (
-        api.endpoints[endpointName] as ApiEndpointQuery<any, any>
-      ).select(arg)(getState())
+
+        return {
+          unwrap() {
+            return initiateOutput.unwrap().then(noop, noop)
```

Comment:
looks like this way you remove information about both fulfilled and rejected values

Reply:
Yeah it makes sense to at least return the output of unwrap